### PR TITLE
[Feature] Support dinov2 backbone

### PR DESCRIPTION
**Pull request by fangyixiao18**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpretrain/models/utils/swiglu_ffn.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    Modified from https://github.com/facebookresearch/dinov2/blob/main/dinov2/layers/swiglu_ffn.py
+    """  # noqa
+
+    def __init__(
+        self,
+        embed_dims: int,
+        feedforward_channels: Optional[int] = None,
+        out_dims: Optional[int] = None,
+        layer_scale_init_value: float = 0.,
+        bias: bool = True,
```
Comment (zzc98) [inline mmpretrain/models/utils/swiglu_ffn.py:23]: It is possible to add an `add_identity` to decide whether to add identity or not , `dropout_layer` or `droprate` to add a dropout layer at the end. Beside, in EVA02, there is a sub_layernorm before w3, this function could be added.

Comment (fangyixiao18) [reply mmpretrain/models/utils/swiglu_ffn.py:25]: updated, please check it

File: mmpretrain/models/utils/attention.py [hunk 2]
```diff
@@ -567,8 +571,10 @@ def __init__(self,
 
         self.out_drop = build_dropout(dropout_layer)
 
-        if use_layer_scale:
-            self.gamma1 = LayerScale(embed_dims)
+        if use_layer_scale or (layer_scale_init_value > 0):
```
Comment (Ezra-Yu) [inline mmpretrain/models/utils/attention.py:574]: I think we can add a Deprecation and deprecate `use_layer_scale` later.

**Review by Ezra-Yu — APPROVED** [review]
LGTM.

**Pull request merged**